Automattic/wp-calypso
# A4A - Partner Directory: upload the agency logo as a wpcom media file

This pull request adds the functionality to upload logos in the Partner directory details form.

<img width="578" alt="Screenshot 2024-07-01 at 10 01 22 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/93091490-9f53-43f5-b2c8-38ffcf098b19">


Resolves: https://github.com/Automattic/automattic-for-agencies-dev/issues/675

## Proposed Changes

* Implement an `A4AImagePicker` component that can be reused with any image upload use cases.
* Implement a Partner directory `LogoPicker` component based on the `A4AImagePicker` component. This component has some partner directory-specific validations.
* Update the `useSubmitForm` hook to handle Logo uploading.


## Testing Instructions

* Use the A4A live link and go to `/partner-directory/agency-details` page
* Fill up the form correctly so all required fields are supplied.
* Test the Company logo picker and confirm it is working as expected.
* Confirm that when submitting the form, the logo image is properly loaded when opening the form again.


## Pre-merge Checklist
